### PR TITLE
callflag: add C#-compliant JSONization, fix #2040

### LINKED ISSUE #2040: Unable to transfer non-native NEP-17 tokens
neo-go is unable to transfer non-native NEP-17 tokens due to an unmarshalling error while processing the JSON retrieved from the RPC method `getcontractstate`

Specifically, an `UnmarshallTypeError` occurs when neo-go tries to unmarshal `nef.tokens.callflags`. It appears that a custom `MarshalJSON`/`UnmarshalJSON` is needed but missing in `pkg/smartcontract/callflag/call_flags.go`. This causes neo-go to exit with the error message `can't fetch matching token from RPC-node: NEP-17 token was not found`.

The native NEP-17 tokens have an empty `nef.tokens` array which is why they can be transferred without error.

This was tested using the NEP-17 token contract 0x0a8b0c0b1981738f468dc630220df18fbb85ff87 (I can provide test tokens if needed for replicating this issue).

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: pkg/smartcontract/callflag/call_flags.go [hunk 1]
```diff
@@ -16,7 +22,93 @@ const (
...
+	var res CallFlag
+
+	for _, flag := range flags {
+		var knownFlag bool
+
+		flag = strings.TrimSpace(flag)
+		for _, f := range basicFlags {
+			if flag == flagString[f] {
+				res |= f
+				knownFlag = true
```
Comment (fyrchik) [inline pkg/smartcontract/callflag/call_flags.go:62]: `break`?

**Review by AnnaShaleva — APPROVED** [review]

File: pkg/smartcontract/callflag/call_flags.go [hunk 2]
```diff
@@ -16,7 +22,93 @@ const (
...
+	WriteStates: "WriteStates",
+	AllowCall:   "AllowCall",
+	AllowNotify: "AllowNotify",
+	States:      "States",
+	ReadOnly:    "ReadOnly",
+	All:         "All",
+	NoneFlag:    "None",
+}
+
+var basicFlags = []CallFlag{ReadOnly, States, ReadStates, WriteStates, AllowCall, AllowNotify}
```
Comment (fyrchik) [inline pkg/smartcontract/callflag/call_flags.go:36]: The order is important for `String` and worth a comment.

**Review by fyrchik — APPROVED** [review]

File: pkg/smartcontract/callflag/call_flags.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/smartcontract/callflag/call_flags.go:66]: Fixed.

File: pkg/smartcontract/callflag/call_flags.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/smartcontract/callflag/call_flags.go:40]: OK, added that.

**Pull request merged**